Automattic/pocket-casts-android
# Adds Builkite test, lint & instrumented test steps

**Please note that this PR will break Github Actions, so we don't want to merge it until the project is complete and we are ready to merge all the PRs. Until then, I'll leave this PR as a draft to avoid accidentally merging it.**

# Description

This PR introduces Buildkite & `fastlane` to the repo while also updating the secrets configuration.

* We are currently using the `Debug` variant for `test` & `lint` Buildkite steps. However, this is likely going to change to `Release` variant. Since it's a simple change, I'd like to postpone this discussion until the more urgent parts of this project are completed.
* The update to secrets configuration includes changes that are necessary for uploading artifacts to Play Store. I've been working on multiple branches up until this point, with lots of commits. I decided to merge relevant commits together for a cleaner history and it made the most sense to have the configuration changes in this PR.
* Similar to the secrets configuration, this PR includes some `.gitignore` changes that are not strictly necessary for this PR. However, there is no harm in ignoring these files early and it makes life easier for me when I switch branches.
* For `fastlane` setup and the initial `build_and_instrumented_test` lane, I followed our approach in WooCommerce-Android. I'll try to do the same for other lanes even when I feel that some things can be marginally improved. I think having a similar approach makes things easier for us and we can always make the improvements in subsequent PRs and improve the projects together. I don't think there is much I'd change in this PR, but I wanted to share this especially with @mokagio as a general approach to the project.
* `fastlane/README.md` is auto-generated and it looks like we are tracking this file in other repositories, so I did the same for this project.

**To test:**

This PR tests itself in Buildkite, so there is not much to test. However, reviewer is always welcome to run the relevant commands to triple check that everything is working as expected in their local environment.

Note that the Buildkite status is not fully reported in the PR. I am running the Buildkite builds manually for now to avoid false negatives in other PRs. You can find the builds here: https://buildkite.com/automattic/pocket-casts-android

# Checklist
